dts/bindings/memory-controllers/atmel,sam-smc.yaml

```diff
@@ -0,0 +1,140 @@
+  cycles. The hold timing isn't directly configurable and needs to be taken into
+  account in the entire cycle time. Here it is set to 0 as the entire cycle
+  time (7) is the sum of the setup (1) and pulse (6) timing.
+  Refer to the connected device's datasheet for specific timing characteristics.
+
+  Finally, in order to make the memory available you will need to define new
+  memory device/s in DeviceTree:
+
+  sram1: sram@60000000 {
+      compatible = "zephyr,memory-region", "mmio-sram";
+      device_type = "memory";
+      reg = <0x60000000 DT_SIZE_K(512)>;
+      zephyr,memory-region = "SRAM1";
+  };
+
+compatible: "atmel,sam-smc"
+
+include: [base.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  "#address-cells":
+    required: true
+    const: 1
+
+  "#size-cells":
+    required: true
+    const: 0
+
+  peripheral-id:
+    type: int
+    description: peripheral ID
+    required: true
+
+child-binding:
+  description: |
+    Child device nodes are representing devices connected to the EBI/SMC bus.
+
+  properties:
+    reg:
+      type: int
+      required: true
+      description: |
+        The device's SMC Chip Select number.
+        Valid range: 0 - 3
+
+    atmel,smc-write-mode:
+      type: string
+      required: true
+      description: |
+        Select which signal is used for the write operation, either the chip
+        select (ncs) or a dedicated write enable pin (nwe). The data is put
+        on the bus during the pulse and hold steps of that signal.
+        The internal data buffers are switched to output mode after the NCS_WR
+        or NWE setup time.
+      enum:
+        - "ncs"
+        - "nwe"
+
+    atmel,smc-read-mode:
+      type: string
+      required: true
+      description: |
+        Select which signal is used for the read operation, either the chip
+        select (ncs) or a dedicated output enable pin (nrd). The SMC samples
+        the read data internally on the rising edge of Master Clock that
+        generates the rising edge of the selected signal.
+      enum:
+        - "ncs"
+        - "nrd"
+
+    atmel,smc-setup-timing:
+      type: array
+      required: true
+      description: |
+        SMC setup timing configurations in cycles for the signals NWE, NCS_WR,
+        NRD and NCS_RD, in that order. Null delay setups are permitted, for
+        consecutive reads/writes.
+        Encoded in 6 bits where the highest bit adds an offset of 128.
+        Effective value for each element: 128 x setup[5] + setup[4:0]
+
+    atmel,smc-pulse-timing:
+      type: array
+      required: true
+      description: |
+        SMC pulse timing configurations in cycles for the signals NWE, NCS_WR,
+        NRD and NCS_RD, in that order. Null pulses are not permitted and lead
+        to unpredictable behavior.
+        Encoded in 7 bits where the highest bit adds an offset of 256.
+        Effective value for each element: 256 x pulse[6] + pulse[5:0]
```

Comment:
This value is used to effectivelly read/write at memory region (pulse phase).
The pulse value is an array of the signals NWE, NCS_WR, NRD and NCS_RD were
each value is configured in terms of MCK cycles and value of 0 is forbiden.
Each value is encoded in 7 bits where the highest bit adds an offset of 256
cycles. The effective value for each element is: 256 x setup[6] + setup[5:0]